encoding.go

```diff
@@ -3,16 +3,39 @@ package encoding
 import (
 	"github.com/signalfx/golib/trace"
 	"time"
+	"strconv"
+	"fmt"
 )
 
+// low, high
+type ID [2]uint64
+
+func GetID(s string) ID {
+	var low, high uint64
+	if len(s) == 16 {
+		low, _ = strconv.ParseUint(s, 16, 64)
+	} else if len(s) == 32 {
+		high, _ = strconv.ParseUint(s[:16], 16, 64)
+		low, _ = strconv.ParseUint(s[16:], 16, 64)
+	}
+	return [2]uint64{low, high}
+}
+
+func (id *ID) String() string {
+	if id[1] == 0 {
+		return fmt.Sprintf("%016x", id[0])
+	}
+	return fmt.Sprintf("%016x%016x", id[1], id[0])
```

Comment:
any reason why the String format has low first and then high?

Reply:
because that's how i wrote it. As long as GetID(s).String() == s, it doesn't matter.  There's a test for it that.